src/librustc_metadata/cstore_impl.rs

```diff
@@ -140,7 +138,27 @@ provide! { <'tcx> tcx, def_id, cdata,
     is_panic_runtime => { cdata.is_panic_runtime(&tcx.dep_graph) }
     is_compiler_builtins => { cdata.is_compiler_builtins(&tcx.dep_graph) }
     has_global_allocator => { cdata.has_global_allocator(&tcx.dep_graph) }
+    is_sanitizer_runtime => { cdata.is_sanitizer_runtime(&tcx.dep_graph) }
+    is_profiler_runtime => { cdata.is_profiler_runtime(&tcx.dep_graph) }
+    panic_strategy => { cdata.panic_strategy(&tcx.dep_graph) }
```

Comment:
Aren't a bunch of these just reading crate-level attributes? Why can't we do that directly through `tcx.get_attrs`?

Reply:
We could, but it would mean that if we allow "free" access, then any change to crate attributes will require recompiling everything. Not sure if that's a problem.

Reply:
Hmm but that could be solved by putting a `Symbol` (for the name of the attribute) in the query, which is what you want most of the time *anyway*.

Reply:
Not all of these are crate attributes as well, some of them, like the panic strategy, can be affected by command line flags as well. The others, sure, but perhaps that's a refactoring for another time?

Reply:
Hmm if there aren't many of them then this PR is probably fine.